### DISCUSSION TITLE
Add property tag into Junit report when running Cucumber tests

### DISCUSSION THREAD

**Discussion by kakarasiri** [cucumber/common#2199] [General]
I would like to add <property> tag into junit report. 
Is it possible for doing that ? or Is there any plugin that we can used ? 

TestRunner.scala 
`@RunWith(classOf[Cucumber])
@CucumberOptions(
  features = Array("classpath:features/"),
  glue = Array("steps"),
  plugin = Array(
    "pretty",
    "html:target/test-reports/cucumber/cucumber.html",
    "json:target/test-reports/cucumber/cucumber.json",
    "junit:target/test-reports/cucumber/cucumber-results.xml",
  ),
)`

Current cucumber-results.xml: 
```
<testsuite>
    <testcase>
        </system-out>
        </system-out>
    </testcase>
</testsuite>
```

What I would like to have: 
```
<testsuite>
    <testcase>
        **<properties>
            <property name="" value=""/>
        </properties>**
        </system-out>
        </system-out>
    </testcase>
</testsuite>
```

**Comment by mpkorstanje** [comment]
What do you mean by "adding a tag to a report"? Do you have a more detailed example? 

Your current example also doesn't seem to match the XSD so it wouldn't be a valid report. Perhaps you can show an example of another framework that produces the desired result?

**Reply by kakarasiri** [reply]
My bad on the typo.

Current cucumber-results.xml:
```
<testsuite>
    <testcase>
        <system-out>
           <![CDATA ... >
        </system-out>
    </testcase>
</testsuite>
```

Example of desired result with Playwright framework: 
```
<testsuite>
    <testcase>
        <properties>
            <property name="TestOwner" value="ABC"> </property>
        </properties>
        <system-out>
            <![CDATA ... >
        </system-out>
    </testcase>
</testsuite>
```


Basically, I would like to log the TestOwner when running cucumber tests into junit report.

**Reply by mpkorstanje** [reply]
The `properties` element is only allowed as part of the `testsuite` element.  Have a look at the XSD. https://github.com/cucumber/junit-xml-formatter/blob/main/jenkins-junit.xsd

How did you create that report with Playwright?

**Reply by kakarasiri** [reply]
It's Jest test not Playwright test. I was wrong at the first place. 

I followed this documentation and able to do it.
Link: https://github.com/jest-community/jest-junit?tab=readme-ov-file#adding-custom-testcase-properties

```
Create a file in your project root directory named junitTestCaseProperties.js:

module.exports = (testResult) => {
  return {
    "dd_tags[test.invocations]": testResult.invocations,
  };
};
```

**Reply by kakarasiri** [reply]
It will render: 
```
<testsuites name="jest tests">
  <testsuite name="addition" tests="1" errors="0" failures="0" skipped="0" timestamp="2017-07-13T09:42:28" time="0.161">
    <testcase classname="addition positive numbers should add up" name="addition positive numbers should add up" time="0.004">
      <properties>
        <property name="dd_tags[test.invocations]" value="1" />
      </properties>
    </testcase>
  </testsuite>
</testsuites>
```

**Reply by mpkorstanje** [reply]
Ah. I see. No that is not possible and I think it is unlikely that we'd support something like that.

I assume you are using Datadog? If so, you may have some luck using JUnit 5 instead of JUnit 4. When using JUnit 5 with the `cucumber-junit-platform-engine` all tags in a Scenario are provided to JUnit 5. So presumably when using Datadog with native instrumentation it can pick them up. 

I haven't tried this myself, so you might want to do a proof of concept first. And if it doesn't work with JUnit 5 I'd suggest requesting this feature from Datadog.

**Reply by kakarasiri** [reply]
Actually, I didn't use Datadog as well. I used jest-junit but it works after follow their steps. 
Does it mean that if I used `cucumber-junit-platform-engine`, I should be able to get all tags in junit.xml as a report as well ?
